### DISCUSSION TITLE
Sequencing different text, rather than just different styles?

### DISCUSSION THREAD

**Discussion by arrowtype** [coldtype/coldtype#117] [Q&A] [answered]
There are some awesome examples of how Coldtype can apply a series of styles/animations to a single piece of text, or glyph-by-glyph in a single piece of text. (For example, [this one](https://github.com/coldtype/coldtype/blob/fad900cfc86e1912cbf301729e11325b9895d510/examples/animations/retails/warping.py) blew my mind! So cool.)

However, I'm having trouble finding any examples of different pieces of text sequenced.

A current idea I’m hoping to try is to animate a piece of text, and cycle through translations in a few languages. Something like (apologies for the Google Translations...):
1. "TYPOGRAPHY IS KINETIC"
2. "ТИПОГРАФИЯ КИНЕТИЧЕСКАЯ"
3. "BIỂU TƯỢNG LÀ KINETIC"

Perhaps my problem is that I don't quite understand how the ASCII keyframes system is plugged into animations, in general. But then, how might that (or something else) be adapted to cycling text?

One thought I had was that I could just put two `@animation` blocks into a file... but that doesn't seem to work, and it would seem pretty repetitive, anyway.

I'll update if I figure out more!

**Comment by arrowtype** [comment]
Or do you tend to just [jump to an NLE](https://coldtype.goodhertz.com/tutorials/animation.html#jumping-to-an-nle) after each tiny cycle of animation?

**Comment by stenson** [comment, marked as answer]
Another great q! I think that advice on jumping to an NLE was written before the AsciiTimeline existed at all, so there really was no good way to choreograph an animation with only code, but now that the AsciiTimeline exists, I would definitely use it for something like what you're describing.

I think something like the below is what I'd go with for a simple animation where the text is changing over time. If you are doing something like a long lyric video though, I'd still say jumping to an NLE is the way to go, though Blender would be my choice atm for that (another coldtype integration that really needs to be documented!)

The "sequencing" here really just comes down to array indexing, where your keyframes are array indexes and you keep the text data in a simple flat array. So really no special behavior for sequencing text, just a generic style for sequencing/timing an array, the key being the `at.current()` line, which will give you the currently "on" item from the sequence.

Then to do the animations on the text as if it was a solo `@animation`, you can use the `current` object instead of the `f` frame object, i.e. `current.e("eeo", 0)` (as below) gives you an easing value for the current timeline item, rather than for the whole animation.

```python
from coldtype import *

txts = [
    "TYPOGRAPHY\nIS KINETIC",
    "ТИПОГРАФИЯ\nКИНЕТИЧЕСКАЯ",
    "BIỂU TƯỢNG\nLÀ KINETIC",
]

at = AsciiTimeline(1, 30, """
                                                <  
0               1               2
""").inflate(lines=[1])

@animation((1400, 540), timeline=at, bg=1, render_bg=1)
def scratch(f):
    current = at.current()
    txt = txts[int(current.name)]

    return (StSt(txt, "SFCompactItalic", 100
        , wght=current.e("eeo", 0))
        .lead(30)
        .xalign(f.a.r)
        .align(f.a.r))

```

https://user-images.githubusercontent.com/89770/182954024-c250ed6d-476d-4caa-bda9-3e11d0f8746c.mp4

You can also tap `v` with the viewer open to toggle a timeline view of an AsciiTimeline, so you'd see this with that code above running:

<img width="812" alt="image" src="https://user-images.githubusercontent.com/89770/182953576-7b7b09c0-921a-4dbc-8d11-8c769f6f9346.png">

Hopefully that answers your question — let me know if not!

**Comment by arrowtype** [comment]
Amazing, this is just what I needed to know! Thank you so much for your helpful answers, and this amazing library. It's exciting to get to know it a little better!
